src/App.js

```diff
@@ -125,7 +125,17 @@ const App = ({ locale, location, dependencies: { appSessionRef, sessionManager }
             exact
             component={ReportsPartialsCampaigns}
           />
-          <PrivateRoute path="/plan-calculator" exact component={PlanCalculator} />
+          <PrivateRoute
+            path={[
+              '/plan-calculator/:typePlanId/buy',
+              '/plan-calculator/:typePlanId/promotions/:promoId/buy',
+              '/plan-calculator/:typePlanId/discounts/:discountId/buy',
+              '/plan-calculator/:typePlanId/promotions/:promoId/discounts/:discountId/buy',
+              '/plan-calculator/:typePlanId/discounts/:discountId/promotions/:promoId/buy',
```

Comment:
It is nice but in my opinion, having a only way (or at least a canonical way) to express a URL is better. 

Remember that the good URL is the URL that does not change, when you have a lot of URLs you should make honor to them.

Also, if promotion and discounts are optional and there is not a hierarchical relation between them, maybe it is better representing them as query string parameters in place of URL segments, ie:

```
/plan-calculator/PlaABC/discounts/VetteranDiscount/promotions/ChristmassPromotion/buy

vs 

/plan-calculator/PlaABC/buy?discount=VetteranDiscount&promotion=ChristmassPromotion
```